packages/store/addon/-private/system/references/reference.ts

```diff
@@ -28,6 +32,8 @@ function isResourceIdentiferWithRelatedLinks(
   return value && value.links && value.links.related;
 }
 
+export const INTERNAL_MODELS = new WeakMap<RecordData, InternalModel>();
```

Comment:
let's key this by reference instance instead of by RecordData, and lets do the same for `recordData` as that would nicely allow us to not leak that property either.

Reply:
We can deprecate accessing `recordData` as part of the same deprecation